frontend/src/component/feature/FeatureView/FeatureOverview/FeatureOverviewEnvironments/FeatureOverviewEnvironment/EnvironmentAccordionBody/StrategyDraggableItem/StrategyItem/StrategyExecution/StrategyExecution.tsx

```diff
@@ -1,372 +1,110 @@
-import { type FC, Fragment, useMemo } from 'react';
-import { Alert, Box, Chip, Link, styled } from '@mui/material';
-import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
-import PercentageCircle from 'component/common/PercentageCircle/PercentageCircle';
-import { StrategySeparator } from 'component/common/StrategySeparator/LegacyStrategySeparator';
+import { Children, isValidElement, type FC, type ReactNode } from 'react';
+import { styled } from '@mui/material';
+import type { CreateFeatureStrategySchema } from 'openapi';
+import type { IFeatureStrategyPayload } from 'interfaces/strategy';
+import { useUiFlag } from 'hooks/useUiFlag';
+import { StrategyExecution as LegacyStrategyExecution } from './LegacyStrategyExecution';
 import { ConstraintItem } from './ConstraintItem/ConstraintItem';
 import { useStrategies } from 'hooks/api/getters/useStrategies/useStrategies';
+import { objectId } from 'utils/objectId';
+import { StrategyEvaluationSeparator } from './StrategyEvaluationSeparator/StrategyEvaluationSeparator';
+import { useCustomStrategyParameters } from './hooks/useCustomStrategyParameters';
+import { useStrategyParameters } from './hooks/useStrategyParameters';
 import { useSegments } from 'hooks/api/getters/useSegments/useSegments';
-import { FeatureOverviewSegment } from 'component/feature/FeatureView/FeatureOverview/FeatureOverviewSegment/FeatureOverviewSegment';
-import { ConstraintAccordionList } from 'component/common/ConstraintAccordion/ConstraintAccordionList/ConstraintAccordionList';
-import {
-    parseParameterNumber,
-    parseParameterString,
-    parseParameterStrings,
-} from 'utils/parseParameter';
-import StringTruncator from 'component/common/StringTruncator/StringTruncator';
-import { Badge } from 'component/common/Badge/Badge';
-import type { CreateFeatureStrategySchema } from 'openapi';
-import type { IFeatureStrategyPayload } from 'interfaces/strategy';
-import { BuiltInStrategies } from 'utils/strategyNames';
+import { SegmentItem } from 'component/common/SegmentItem/SegmentItem';
 
-interface IStrategyExecutionProps {
-    strategy: IFeatureStrategyPayload | CreateFeatureStrategySchema;
-    displayGroupId?: boolean;
-}
+const StyledGrayscale = styled('div', {
+    shouldForwardProp: (prop) => prop !== 'enabled',
+})<{ enabled: boolean }>(({ enabled }) =>
+    enabled ? { filter: 'grayscale(1)', opacity: 0.67 } : {},
```

Comment:
Just make sure the rendered text still has enough contrast against the background to pass AA validation